From 8fdf42b87a407409f42acff083e93bfdd586aa1a Mon Sep 17 00:00:00 2001
From: Ricardo Cerqueira <ricardo@cyngn.com>
Date: Fri, 6 Nov 2015 01:35:34 +0000
Subject: [PATCH 18/25] legacy: Add back old fdAlbum rule

* facedetection still uses a file at /data :(
* Partial pick from commit 3aaeeceb270dc6c8af8dd9a95fa8b9e33097ff50
  and squash of 342476ec817c8bc371d1716f39dec11cb2ded352

Change-Id: Ifa500ca57dc71831074a39fb5b05246c12cd0d4c
---
 legacy-common/file_contexts  | 3 +++
 legacy-common/mm-qcamerad.te | 4 ++++
 2 files changed, 7 insertions(+)

diff --git a/legacy-common/file_contexts b/legacy-common/file_contexts
index 88b5882..ec55bdd 100644
--- a/legacy-common/file_contexts
+++ b/legacy-common/file_contexts
@@ -32,3 +32,6 @@
 
 # Subsys restart
 /sys/devices(/platform)?/soc\.0/[a-z0-9\.:]+,[a-z0-9\-]+/subsys[0-9]+/name    u:object_r:sysfs_ssr:s0
+
+# Camera
+/data/fdAlbum                                                       u:object_r:camera_data_file:s0
diff --git a/legacy-common/mm-qcamerad.te b/legacy-common/mm-qcamerad.te
index 87ce39a..951dbe2 100644
--- a/legacy-common/mm-qcamerad.te
+++ b/legacy-common/mm-qcamerad.te
@@ -5,3 +5,7 @@ allow mm-qcamerad camera_data_file:sock_file { create unlink };
 
 allow mm-qcamerad system_server:unix_stream_socket rw_socket_perms;
 allow mm-qcamerad sensorservice_service:service_manager find;
+
+# /data/fdAlbum
+type_transition mm-qcamerad system_data_file:file camera_data_file "fdAlbum";
+allow mm-qcamerad camera_data_file:file create_file_perms;
-- 
2.7.4

